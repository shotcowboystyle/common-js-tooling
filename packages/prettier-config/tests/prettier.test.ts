// @ts-expect-error TS considers the src to not be a module because it uses module.exports but vitest transpiles it just fine for this
import prettierConfig from '../src';

describe('Prettier Config', () => {
  test('should export rules', () => {
    expect(prettierConfig.$schema).toBe('http://json.schemastore.org/prettierrc');
    expect(prettierConfig.useTabs).toBe(true);
    expect(prettierConfig).toMatchSnapshot();
  });
});
