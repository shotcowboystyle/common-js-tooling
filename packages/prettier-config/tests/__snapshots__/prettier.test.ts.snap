// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Prettier Config > should export rules 1`] = `
{
  "$schema": "http://json.schemastore.org/prettierrc",
  "endOfLine": "lf",
  "overrides": [
    {
      "files": "*.yml",
      "options": {
        "tabWidth": 2,
        "useTabs": false,
      },
    },
  ],
  "printWidth": 150,
  "quoteProps": "as-needed",
  "semi": true,
  "singleQuote": true,
  "tabWidth": 4,
  "trailingComma": "none",
  "useTabs": true,
}
`;
